### PR TITLE
[primTorch] decompositions for upsample_bicubic2d

### PR DESCRIPTION
**Pull request by fdrocha**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #85403
* #85402

FYI, this decomposition seems to be significantly slower than the lowering in torchinductor:

```
------------------------------------- upsample_bicubic2d -------------------------------------]
                                                              |  lowering  |  Inductor  |  Eager
32 threads: ------------------------------------------------------------------------------------
      (torch.Size([16, 4, 128, 256]),), ((512, 1024), True)   |    1.8     |   3.880    |   1.4
      (torch.Size([16, 4, 128, 256]),), ((512, 1024), False)  |    1.9     |   3.887    |   1.4
```

This seems related to the fact that in the lowering we can use int32s as the indices and in the decomp we can only use int64s (see https://github.com/pytorch/torchdynamo/issues/1293).

### CODE REVIEW

File: torch/_decomp/decompositions.py [hunk 1]
```diff
@@ -2030,3 +2033,89 @@ def binary_cross_entropy_with_logits(
...
+            return output_size
+        else:
+            return tuple(
+                int(w * scale) for w, scale in zip(input_size[2:], scale_factors)
+            )
+
+    output_size = compute_output_size(a.shape, output_size, scale_factors)
+    scale_h = scale_factors[0] if scale_factors else None
+    scale_w = scale_factors[1] if scale_factors else None
+    return upsample_bicubic2d_default(a, output_size, align_corners, scale_h, scale_w)
```
Comment (lezcano) [inline torch/_decomp/decompositions.py:2121]: It's not clear why this decomposition is necessary, given that this function is `CompositeExplicit`.

Comment (fdrocha) [reply torch/_decomp/decompositions.py:2121]: I just filled an issue pertaining to this, see https://github.com/pytorch/pytorch/issues/85405

**Comment by fdrocha** [comment]
@ngimel, @ezyang could you PTAL? Mostly this was discussed in other PRs and issues

**Review by ngimel — APPROVED** [review]
lgtm, thanks

**Pull request closed without merging**